website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,462 @@
+    // Assigin the texture coordinates without any modification to be recived in the fragment
+    frag_texCoords = aTextureCoord;
+}";
+```
+in this case, we are putting manually the `location` of `aTextureCoord` as 1, because we already have a attribute in the location 0.
+
+
+Fragment Shader:
+```
+#version 330 core
+
+// Receive the input from the vertex shader in an attribute
+in vec2 frag_texCoords;
+
+out vec4 out_color;
+
+void main()
+{
+    // This will allow us to see the texture coordinates in action!
+    out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0.0, 1.0);
+}
+```
+As you already know, the Vertex Shader runs for every vertex of the mesh and the Fragment Shader runs for every pixel inside the mesh.
+This conversion looks weird at first glance, but the data communication between vertex and fragment shader are surely possible because
+OpenGL automatically interpolates the data outputted by the Vertex Shader when it's received by the Fragment Shader.
+
+Now we just need to assign the correct layout for `aTextureCoord`. You should add these lines below our first `VertexAttribPointer` call:
+
+```c#
+const uint texCoordLoc = 1;
+_gl.EnableVertexAttribArray(texCoordLoc);
+_gl.VertexAttribPointer(texCoordLoc, 2, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)(3 * sizeof(float)));
+```
+Here we're specifying a new vertex attribute, which consists of two floats and repeats every 20 bytes (`5 * sizeof(float)` = 20,
+that's the size of one vertex).
+
+Pay attention to the last parameter, the pointer! It represents the number of bytes by which the offset within the vertex buffer
+should be incremented before reading the vertex attribute data. This means that for this attribute OpenGL will advance past first
+3 floats (`3 * sizeof(float)`) worth of data (which are our vertex XYZ coordinates) and instead read the next 2 floats after that.
+
+If you do everything right, you will see this result!
+
+![Quad with UVs](../../../images/opengl/chapter1/lesson3/quad-with-uvs.png)
+
+But we aren't there just yet. We can't see any texture in our quad!
+What you are seeing is the texture coordinates we passed displayed as a color!
+
+The Red, Green and mix between these two colors you see are, respectively, the X and Y texture coordinate values of that pixel.
+In the shader, the values are being read like this:
+
+![UV values](../../../images/opengl/chapter1/lesson3/quad-with-uvs-and-numbers.png)
+
+As you can see, the higher the X/U coordinate, more reddish is the pixel, and the higher the Y/V coordinate, the more greenish the pixel.
+Even though we only specified UV values for each vertex, all pixels in the quad have UV values! That's because, as you read before,
+the fragment shader interpolates the coordinates for us, saving us a lot of work on our end.
+If we follow the diagonal from (0, 0) to (1, 1), the amount of red and green in the output colour increases at the same rate, resulting in a yellow color.
```

Comment:
I think reddish is worse as it just makes me think of raddish haha.
```suggestion
As you can see, as the X/U coordinate increases so does the amount of red in the output pixel, and likewise as the Y/V coordinate increases so does the amount of green.
Even though we only specified UV values for each vertex, all pixels in the quad have UV values. That's because, as you read before,
the fragment shader interpolates the coordinates for us, saving us a lot of work on our end.
If we follow the diagonal from (0, 0) to (1, 1), the amount of red and green in the output colour increases at the same rate, resulting in a yellow color.
```

Reply:
((Replace `colour` with `color` as the repo follows Freedom English))

Reply:
Ugh, yes, against my better judgement. Will update my comment.

Reply:
😈

Reply:
This suggestion has been applied, I'm marking it as resolved (ping @Perksey)